packages/devtools-network-console/src/reducers/response/index.ts

```diff
@@ -45,11 +45,21 @@ export default function reduceResponse(collection: ResponsesState = DEFAULT_RESP
             break;
 
         case 'RESPONSE_END_REQUEST':
+            let isWebsocketUpgrade = false;
```

Comment:
this is the logic to determine if WS panel should be added to the response viewer. Its possible we could move this to the WS reducer, but we would need to update the responseViewer tsx UI to be hooked up to the WS state as well

Reply:
If you need to hook up a .tsx to a Store but don't want to go through the process of using `mapStateToProps`, you can use the react-redux hook `useSelector`, e.g.:

    const isWSUpgrade = useSelector<IView>(store => {
      const req = store.websockets.get(props.requestId);
      return !!req;
    });

Reply:
Thanks! i'll try this out tomorrow!